cpp/src/io/comp/uncomp.cpp

```diff
@@ -353,12 +353,7 @@ gdf_error io_uncompress_single_h2d(const void *src, gdf_size_type src_size, void
     {
         uncomp_len = comp_len * 4 + 4096; // In case uncompressed size isn't known in advance, assume ~4:1 compression for initial size
     }
-    cu_err = cudaMallocManaged((void **)&uncomp_data, uncomp_len);
-    if (cu_err != cudaSuccess)
-    {
-        gdf_err = GDF_MEMORYMANAGER_ERROR;
-        goto error_exit;
-    }
+    uncomp_data = new uint8_t[uncomp_len];
```

Comment:
You've switched from allocating with `cudaMallocManaged` to using `new`, but you're still deallocating with `cudaFree` in the error case. 

https://github.com/rapidsai/cudf/blob/f40ecfd0203ec4b96324bd61dbd4a10c4b9523ac/cpp/src/io/comp/uncomp.cpp#L384

In the success case, you set `*dst = uncomp_data;`. Was the code that uses `dst` also updated to eventually free this memory correctly (i.e., not `cudaFree`)?

All in all, I'm uncomfortable with this style of allocating memory since it is so prone to memory leaks. Could you pass in `dst` as `std::vector<uint8_t>& dst` instead? Then you could just call `dst.resize(ucomp_len)` inside this function to allocate the necessary memory and you eliminate the chance for a memory leak.